### PR TITLE
DMA protection feature

### PR DESCRIPTION
**Pull request by Stojek139808**

### CODE REVIEW

File: docs/dasharo-menu-docs/dasharo-system-features.md [hunk 1]
```diff
@@ -19,6 +19,19 @@ This menu offers security-sensitive options like:
   BIOS flash. Recommended to be enabled if [Firmware setup password](overview.md#user-password-management)
   is set. Disable the protection if you need access to flash with flashrom.
 
+- `Dma protection` - Early Boot DMA protection is a feature that utilizes the
+  IOMMU early in the boot process to prevent rogue DMA-capable devices such as
+  PCIe add-in cards and USB 3 devices from access to memory. This prevents a
+  class of DMA attacks that allow for exfiltration of secrets and installation
+  of malware early in the boot process. DMA protection can be enabled in two
+  ways:
```
Comment (macpijan) [inline docs/dasharo-menu-docs/dasharo-system-features.md:27]: @Stojek139808 Something is missing here? Or `IOMMU handoff at ExitBootServices` is supposed to be one of these options?

File: docs/dasharo-menu-docs/dasharo-system-features.md [hunk 2]
```diff
@@ -19,6 +19,19 @@ This menu offers security-sensitive options like:
...
+- `IOMMU handoff at ExitBootServices` - this option is available when
+  `Dma protection` is enabled. It changes the moment when IOMMU is re-enabled.
+  If this option is selected, the it's re-enabled at ExitBootService when
```
Comment (macpijan) [inline docs/dasharo-menu-docs/dasharo-system-features.md:30]:
```suggestion
  If this option is selected, then it's re-enabled at ExitBootService when
```

Comment (macpijan) [reply docs/dasharo-menu-docs/dasharo-system-features.md:30]: `ExitBootService` or `ExitBootService` ? The description in the first line of this feature is different

**Review by macpijan — CHANGES_REQUESTED** [review]

File: docs/dasharo-menu-docs/dasharo-system-features.md (hunk 1 above)
Comment (Stojek139808) [reply docs/dasharo-menu-docs/dasharo-system-features.md]: It is one of the options, but it sounded bad so I just changed it.

File: docs/dasharo-menu-docs/dasharo-system-features.md (hunk 2 above)
Comment (Stojek139808) [reply docs/dasharo-menu-docs/dasharo-system-features.md]: It's `ExitBootServices`, fixed the other accordingly.

File: docs/dasharo-menu-docs/dasharo-system-features.md [hunk 3]
```diff
@@ -19,6 +19,18 @@ This menu offers security-sensitive options like:
   BIOS flash. Recommended to be enabled if [Firmware setup password](overview.md#user-password-management)
   is set. Disable the protection if you need access to flash with flashrom.
 
+- `Dma protection` - Early Boot DMA protection is a feature that utilizes the
+  IOMMU early in the boot process to prevent rogue DMA-capable devices such as
+  PCIe add-in cards and USB 3 devices from access to memory. This prevents a
+  class of DMA attacks that allow for exfiltration of secrets and installation
+  of malware early in the boot process.
+- `IOMMU handoff at ExitBootServices` - this option is available when
```
Comment (miczyg1) [inline docs/dasharo-menu-docs/dasharo-system-features.md:27]: I still have suggestions to the format of this option. Keep the comment here until it is applied.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/dasharo-menu-docs/dasharo-system-features.md [hunk 4]
```diff
@@ -19,6 +19,18 @@ This menu offers security-sensitive options like:
...
+  `Dma protection` is enabled. It changes the moment when IOMMU is re-enabled.
+  If this option is selected, then it's re-enabled at ExitBootServices when
```
Comment (miczyg1) [inline docs/dasharo-menu-docs/dasharo-system-features.md:29]: The thing is that the IOMMu is being disabled when handing-off to OS to not cause problems with PCI enumeration. OS should set up IOMMU per its own needs during the handoff state. The option here is to mitigate problems with OSes that do not work well with EBS handoff.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/dasharo-menu-docs/dasharo-system-features.md [hunk 5]
```diff
@@ -19,6 +19,18 @@ This menu offers security-sensitive options like:
   BIOS flash. Recommended to be enabled if [Firmware setup password](overview.md#user-password-management)
   is set. Disable the protection if you need access to flash with flashrom.
 
+- `Early boot DMA Protection` - Enables IOMMU DMA protection early during POST.
+  DMA protection is a feature that utilizes the IOMMU early in the boot process
+  to prevent rogue DMA-capable devices such as PCIe add-in cards and USB 3
+  devices from access to memory. This prevents a class of DMA attacks that allow
```
Comment (miczyg1) [inline docs/dasharo-menu-docs/dasharo-system-features.md:25]: USB? If anything USB4 or Thunderbolt

File: docs/dasharo-menu-docs/dasharo-system-features.md [hunk 6]
```diff
@@ -19,6 +19,18 @@ This menu offers security-sensitive options like:
...
+  for exfiltration of secrets and installation of malware early in the boot
+  process.
+
+- `Keep IOMMU enabled when transfer control to OS` - (only available when
+  `Early boot DMA Protection` is selected) Keeps IOMMU DMA protection enabled
+  during ExitBootServices when control is passed to the OS. If unsure or see any
+  problems with DMA/IOMMU during OS boot, keep it disabled.
```
Comment (miczyg1) [inline docs/dasharo-menu-docs/dasharo-system-features.md:32]:
```suggestion
  problems with DMA/IOMMU during OS boot, keep it disabled.
  NOTE: Windows may not boot if the option is enabled.
```

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: docs/dasharo-menu-docs/dasharo-system-features.md (hunk 5 above)
Comment (miczyg1) [reply docs/dasharo-menu-docs/dasharo-system-features.md:25]: `such as PCIe add-in cards and USB 3` -> `such as PCIe add-in cards and USB4/Thunderbolt`

Comment (Stojek139808) [reply docs/dasharo-menu-docs/dasharo-system-features.md]: Updated the description.

**Review by miczyg1 — APPROVED** [review]

**Pull request merged**